src/core/desktopentry.cpp

```diff
@@ -495,6 +507,21 @@ void DesktopEntryManager::onScanCompleted(const QList<ParsedDesktopEntryData>& s
 	auto newLowercaseEntries = QHash<QString, DesktopEntry*>();
 
 	for (const auto& data: scanResults) {
+		auto lowerId = data.id.toLower();
+
+		if (data.hidden) {
+			if (auto* victim = newEntries.take(data.id)) victim->deleteLater();
+			newLowercaseEntries.remove(lowerId);
+
+			if (auto it = oldEntries.find(data.id); it != oldEntries.end()) {
+				it.value()->deleteLater();
+				oldEntries.erase(it);
+			}
+
+			qCDebug(logDesktopEntry) << "Masking hidden desktop entry" << data.id;
+			continue;
+		}
```

Comment:
1. Why deleteLater?
2. Is hidden supposed to match only things lower than it in the path, or things higher than it as well? Because this looks like its only matching things lower than it.